sdk/attestation/attestation/src/attestationClient.ts

```diff
@@ -99,17 +142,114 @@ export class AttestationClient {
+    const { span, updatedOptions} = createSpan("AttestationClient-attestOpenEnclave", options);
+    try {
+      const attestationResult = await this._client.attestation.attestOpenEnclave({
+        report: report,
+        initTimeData: (options.initTimeData ? { 
+          data: options.initTimeData.data,
+          dataType: (options.initTimeData.isJson ? "JSON" : "Binary"),
+        } : undefined),
+        runtimeData:  (options.runTimeData ? { 
+          data: options.runTimeData.data,
+          dataType: (options.runTimeData.isJson ? "JSON" : "Binary"),
+        } : undefined),
+        draftPolicyForAttestation: (options.draftPolicyForAttestation ?? undefined),
+      }, updatedOptions);
+
+      let token = new AttestationToken(attestationResult.token ?? "");
+      const serializer = new Serializer({ AttestationResultMapper });
+
+      let result = new AttestationResponse<AttestationResult>(token, serializer, AttestationResultMapper, "attestationResult", attestationResult);
+      return result;
+
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message});
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /** Attests a quote generated from SGX Enclave using the Intel SDK.
+   * 
+   * @param quote An SGX quote generated by an SGX enclave.
+   * @param options Operation options for the attestOpenEnclave API call.
+   * @returns 
+   */
+   public async attestSgxEnclave(quote : Uint8Array, options : AttestSgxEnclaveOptions={}) : Promise<AttestationResponse<AttestationResult>> {
+    const { span, updatedOptions} = createSpan("AttestationClient-attestSgxEnclave", options);
+    try {
+      const attestationResult = await this._client.attestation.attestSgxEnclave({
+        quote: quote,
+        initTimeData: (options.initTimeData ? { 
+          data: options.initTimeData.data,
+          dataType: (options.initTimeData.isJson ? "JSON" : "Binary"),
+        } : undefined),
+        runtimeData:  (options.runTimeData ? { 
+          data: options.runTimeData.data,
+          dataType: (options.runTimeData.isJson ? "JSON" : "Binary"),
+        } : undefined),
+        draftPolicyForAttestation: (options.draftPolicyForAttestation ?? undefined),
+      }, updatedOptions);
+
+      let token = new AttestationToken(attestationResult.token ?? "");
+      const serializer = new Serializer({ AttestationResultMapper });
+
+      let result = new AttestationResponse<AttestationResult>(token, serializer, AttestationResultMapper, "attestationResult", attestationResult);
+      return result;
+
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message});
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+
+  /** Attest a TPM based enclave.
+
+   * See the `TPM Attestation Protocol Reference <https://docs.microsoft.com/en-us/azure/attestation/virtualization-based-security-protocol>`_ for more information.
```

Comment:
Let's use the link tag for this url: https://tsdoc.org/pages/tags/link/. You can refer to this page for a list of useful tags to use.